mailbox/store/src/main/java/org/apache/james/mailbox/store/AnnotationKeyValidator.java

```diff
@@ -0,0 +1,60 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.store;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.james.mailbox.exception.MailboxException;
+
+import com.google.common.base.CharMatcher;
+
+public class AnnotationKeyValidator {
+    public static final CharMatcher NAME_ANNOTATION_PATTERN = CharMatcher.ASCII
+                                                      .and(CharMatcher.inRange('\u0000', '\u0019').negate())
+                                                      .and(CharMatcher.isNot('*'))
+                                                      .and(CharMatcher.isNot('%'));
+
+    public static final String SLASH_CHARACTER = "/";
+
+    public static final String TWO_SLASH_CHARACTER = "//";
+
+    public static boolean isValid(String input) throws MailboxException {
+        if (input == null) {
```

Comment:
Preconditions.checkNotNull throws NPE in case of null but I don't want to catch more Exception than my expected exception (InvalidAnnotationException). That's why I did not use the Preconditions.